Enterprise-CMCS/macpro-mdct-mcr
# MDCT-2088 - Server-side validation failure should notify user

## Description
[MDCT-2088](https://qmacbis.atlassian.net/browse/MDCT-2088) I have some reservations here and would like _your_ input.

With what is in this PR you do get an error banner whenever it fails. However,
1. It does not go away until you refresh the page
2. It does not tell you what happened (this is sometimes desired though)
3. It does not save the values you filled in. It's just lost and you have to re-enter. Very sad.
4. Autosave will change some or most of the above feel and behavior
5. I tried putting a condition on `StandardReportPage`'s `onSubmit` such that it won't navigate if `errorMessage` is set. This sets up a race condition between the state setter and the navigate. From what I saw the navigator won every time - meaning it'll let you navigate one page but then subsequent navigations blocks.

Wondering if we want to just log the error for support and not show it? I still think that's bad because people might not notice stuff isn't getting saved. The alternative though is a bit complicated. 

Ok thoughts time please comment.

### How to test
Comment out lines 52, 62, 73, and 84 of `services/ui-src/src/components/reports/ReportProvider.tsx`.

Add a `throw new Error("test error")` to line 31 of `services/app-api/handlers/reports/update.ts`
Try to submit a report. See the error banner and the associated issues mentioned above (Note #4 will not happen unless you add that condition)

If you navigate around you'll notice that the banner doesn't go away. 

Now uncomment lines 52, 62, 73, and 84 of `services/ui-src/src/components/reports/ReportProvider.tsx`.

If you try and submit the form, you'll see that the error banner still appears but as you navigate around the form, it will eventually get removed.


## Code author checklist

## Reviewer checklist (two different people)
